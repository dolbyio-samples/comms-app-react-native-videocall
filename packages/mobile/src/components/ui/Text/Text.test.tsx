import { defaultTheme as theme } from '@dolbyio/comms-uikit-common';
import React from 'react';
import { Text } from 'react-native';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';

import { render, mergeStyles } from '../../../utils/test-utils';

import CustomText from './Text';

const { colors } = theme;

const text = 'Sample text';
const testID = 'testID';
const color = 'primary.500';
const customComponent = <Text testID="say-hello">Hello World</Text>;

describe('Text component', () => {
  test('Renders text passed as children', () => {
    const { getByText } = render(<CustomText>{text}</CustomText>);
    const regEx = new RegExp(text, 'i');
    expect(getByText(regEx)).not.toBeNull();
  });

  test('Passes TestID', () => {
    const { getByTestId } = render(<CustomText testID={testID}>{text}</CustomText>);
    expect(getByTestId(testID)).not.toBeNull();
  });

  test('Passes given styles', () => {
    const align = 'center';
    const { getByTestId } = render(
      <CustomText testID={testID} color={color} type="H3" align={align}>
        {text}
      </CustomText>,
    );
    const element = getByTestId(testID);
    const styles = mergeStyles(element.props.style);
    expect(styles.color).toBe(colors.primary[500]);
    expect(styles.textAlign).toBe(align);
  });

  test('Handles uppercase on/off', () => {
    const { getByTestId, rerender } = render(<CustomText testID={testID}>{text}</CustomText>);
    const element = getByTestId(testID);
    let styles = mergeStyles(element.props.style);
    expect(styles.textTransform).toBeUndefined();
    rerender(
      <CustomText testID={testID} uppercase>
        {text}
      </CustomText>,
    );
    styles = mergeStyles(element.props.style);
    expect(styles.textTransform).toBe('uppercase');
    // validate default align is left
    expect(styles.textAlign).toBe('left');
  });
});

describe('Text component snapshot', () => {
  test('should match default param configuration', () => {
    const tree = renderer.create(<CustomText testID={testID}>{text}</CustomText>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should match with children component', () => {
    const tree = renderer
      .create(
        <CustomText testID={testID} uppercase align="right" type="bodySmallSemiBold">
          {customComponent}
        </CustomText>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
