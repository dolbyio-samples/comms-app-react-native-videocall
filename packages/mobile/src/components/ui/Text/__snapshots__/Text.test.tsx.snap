// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component snapshot should match default param configuration 1`] = `
<Text
  style={
    Array [
      Object {
        "letterSpacing": 1.22,
      },
      Object {
        "fontSize": 15,
        "fontWeight": "600",
        "lineHeight": 24,
      },
      false,
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Roboto",
        "textAlign": "left",
      },
    ]
  }
  testID="testID"
>
  Sample text
</Text>
`;

exports[`Text component snapshot should match with children component 1`] = `
<Text
  style={
    Array [
      Object {
        "letterSpacing": 1.22,
      },
      Object {
        "fontSize": 13,
        "fontWeight": "700",
        "lineHeight": 16,
      },
      Object {
        "textTransform": "uppercase",
      },
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Roboto",
        "textAlign": "right",
      },
    ]
  }
  testID="testID"
>
  <Text
    testID="say-hello"
  >
    Hello World
  </Text>
</Text>
`;
